src/components/InputField.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+
+export const InputField = () => {
+  const [textField, setTextField] = useState<string>('');
+
+  const handleClick = () => {
+    setTextField('');
+  };
+
+  return (
+    <div className="relative w-64">
+      <input
+        type="text"
+        className="w-full px-4 py-2 border border-gray-300 rounded-md focus:outline-none focus:border-blue-500"
+        onChange={(e) => {
+          setTextField(e.target.value);
+        }}
+        value={textField}
+      />
+      {textField && (
+        <button
+          className="absolute inset-y-0 right-0 flex items-center pr-3 text-gray-500 hover:text-gray-700"
+          onClick={handleClick}
+        >
+          ×
+        </button>
```

Comment:
The type attribute for <button> elements is "submit" by default. 

However, since this button is meant to be clearing input values, you should add `type=button` to avoid unintended form submissions.

More on this here: https://www.w3schools.com/TAGs/att_button_type.asp